vulkan_helpers/shader_collection.h

```diff
@@ -0,0 +1,74 @@
+/* Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef VULKAN_HELPERS_SHADER_COLLECTION_H_
+#define VULKAN_HELPERS_SHADER_COLLECTION_H_
+
+#include <cassert>
+
+namespace vulkan {
+
+// This class can be used to multiplex between different shaders:
+// GLSL shaders compiled to SPIR-V through glslc
+// HLSL shaders compiled to SPIR-V through glslc
+// HLSL shaders compiled to SPIR-V through DXC
+// The class can return appropriate shaders based on the given 'shader_compiler'
+// configuration: shader-compiler={dxc-hlsl, glslc-glsl, glslc-hlsl}
+class ShaderCollection {
+ public:
+  using u32vec = std::vector<uint32_t>;
+  ShaderCollection(const char* shader_compiler,
+                   u32vec& glslc_glsl_vertex_shader,
+                   u32vec& glslc_glsl_fragment_shader,
+                   u32vec& glslc_hlsl_vertex_shader,
+                   u32vec& glslc_hlsl_fragment_shader,
+                   u32vec& dxc_hlsl_vertex_shader,
+                   u32vec& dxc_hlsl_fragment_shader) {
+    // Find out the shader compiler
+    if (strncmp(shader_compiler, "glslc-glsl", 10) == 0) {
+      vertex_shader = glslc_glsl_vertex_shader.data();
+      fragment_shader = glslc_glsl_fragment_shader.data();
+      vertex_shader_word_count = glslc_glsl_vertex_shader.size();
+      fragment_shader_word_count = glslc_glsl_fragment_shader.size();
+    } else if (strncmp(shader_compiler, "glslc-hlsl", 10) == 0) {
+      vertex_shader = glslc_hlsl_vertex_shader.data();
+      fragment_shader = glslc_hlsl_fragment_shader.data();
+      vertex_shader_word_count = glslc_hlsl_vertex_shader.size();
+      fragment_shader_word_count = glslc_hlsl_fragment_shader.size();
+    } else if (strncmp(shader_compiler, "dxc-hlsl", 8) == 0) {
+      vertex_shader = dxc_hlsl_vertex_shader.data();
+      fragment_shader = dxc_hlsl_fragment_shader.data();
+      vertex_shader_word_count = dxc_hlsl_vertex_shader.size();
+      fragment_shader_word_count = dxc_hlsl_fragment_shader.size();
+    } else {
+      assert(false && "The given shader_compiler is invalid.");
```

Comment:
I would prefer it if we passed down the log object, and had a  LOG_ASSERT() here, a normal assert() will not show up properly in Android builds.